index.d.ts

```diff
@@ -1,426 +1,339 @@
+  }
 
-type CaptureBody = 'off' | 'errors' | 'transactions' | 'all';
-type CaptureErrorLogStackTraces = 'never' | 'messages' | 'always';
-type LogLevel = 'trace' | 'debug' | 'info' | 'warn' | 'warning' | 'error' | 'fatal' | 'critical' | 'off';
+  // Span API
+  // https://www.elastic.co/guide/en/apm/agent/nodejs/current/span-api.html
+  export interface Span extends GenericSpan {
+    // The following properties and methods are currently not documented as their API isn't considered official:
+    // customStackTrace(), setDbContext()
 
-type CaptureErrorCallback = (err: Error | null, id: string) => void;
-type FilterFn = (payload: Payload) => Payload | boolean | void;
-type LabelValue = string | number | boolean | null | undefined;
-type KeyValueConfig = string | Labels | Array<Array<LabelValue>>
+    transaction: Transaction;
+    name: string;
 
-type Payload = { [propName: string]: any }
+    end (endTime?: number): void;
 
-type PatchHandler = (exports: any, agent: Agent, options: PatchOptions) => any;
+    setOutcome(outcome: Outcome): void;
+  }
 
-interface PatchOptions {
-  version: string | undefined,
-  enabled: boolean
-}
+  // https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html
+  export interface AgentConfigOptions {
+    abortedErrorThreshold?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    active?: boolean;
+    addPatch?: KeyValueConfig;
+    apiKey?: string;
+    apiRequestSize?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    apiRequestTime?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    asyncHooks?: boolean;
+    breakdownMetrics?: boolean;
+    captureBody?: CaptureBody;
+    captureErrorLogStackTraces?: CaptureErrorLogStackTraces;
+    captureExceptions?: boolean;
+    captureHeaders?: boolean;
+    captureSpanStackTraces?: boolean;
+    cloudProvider?: string;
+    configFile?: string;
+    containerId?: string;
+    disableInstrumentations?: string | string[];
+    disableSend?: boolean;
+    environment?: string;
+    errorMessageMaxLength?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    errorOnAbortedRequests?: boolean;
+    filterHttpHeaders?: boolean;
+    frameworkName?: string;
+    frameworkVersion?: string;
+    globalLabels?: KeyValueConfig;
+    hostname?: string;
+    ignoreMessageQueues?: Array<string>;
+    ignoreUrls?: Array<string | RegExp>;
+    ignoreUserAgents?: Array<string | RegExp>;
+    instrument?: boolean;
+    instrumentIncomingHTTPRequests?: boolean;
+    kubernetesNamespace?: string;
+    kubernetesNodeName?: string;
+    kubernetesPodName?: string;
+    kubernetesPodUID?: string;
+    logLevel?: LogLevel;
+    logUncaughtExceptions?: boolean;
+    logger?: PinoLogger | Logger;
```

Comment:
REVIEWER NOTE: This is an independent change that could be done in a separate commit. A while back we added support for the given logger being a pino logger instance.